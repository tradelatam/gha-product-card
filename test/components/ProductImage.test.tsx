import React  from 'react';
import { ProductImage } from "../../src/components/ProductImage";
import renderer  from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { product2 } from './data/products';

describe('Pruebas en ProductImage', () => {  

    test('debe de mostrar el componente correctamente con la imagen personalizada', () => {  

        const wrapper = renderer.create(
            <ProductImage img="https://www.cleverfiles.com/howto/wp-content/uploads/2018/03/minion.jpg" className="custon-class" />
        )
        //console.log( wrapper.toJSON() );
        expect( wrapper.toJSON() ).toMatchSnapshot();
    });

    test('debe de mostrar el componente con la imagen que viene en el producto', () => {  

        const wrapper = renderer.create(
            <ProductCard product={ product2 }>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

})