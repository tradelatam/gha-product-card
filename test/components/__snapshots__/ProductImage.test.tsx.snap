// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage debe de mostrar el componente con la imagen que viene en el producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product imagen"
    className="productImg undefined"
    src="./Coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product imagen"
  className="productImg custon-class"
  src="https://www.cleverfiles.com/howto/wp-content/uploads/2018/03/minion.jpg"
/>
`;
